compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -2767,13 +2782,44 @@ TR::AMD64RegImm64Instruction::addMetaDataForCodeAddress(uint8_t *cursor)
                                          __FILE__, __LINE__, getNode());
                   }
                break;
-            case TR_JNIStaticTargetAddress:
-            case TR_JNISpecialTargetAddress:
-            case TR_JNIVirtualTargetAddress:
             case TR_StaticRamMethodConst:
             case TR_VirtualRamMethodConst:
             case TR_SpecialRamMethodConst:
-               cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor, (uint8_t *) getNode()->getSymbolReference(), getNode() ? (uint8_t *)(intptr_t)getNode()->getInlinedSiteIndex() : (uint8_t *)-1,  (TR_ExternalRelocationTargetKind) _reloKind, cg()),  __FILE__,__LINE__, getNode());
+               cg()->addExternalRelocation(
+                  new (cg()->trHeapMemory()) TR::ExternalRelocation(
+                                                cursor,
+                                                (uint8_t *) getNode()->getSymbolReference(),
+                                                getNode()
+                                                   ? (uint8_t *)(intptr_t)getNode()->getInlinedSiteIndex()
+                                                   : (uint8_t *)-1,
+                                                (TR_ExternalRelocationTargetKind) _reloKind,
+                                                cg()),
+                     __FILE__,__LINE__, getNode());
+               break;
+            case TR_JNIStaticTargetAddress:
+            case TR_JNISpecialTargetAddress:
+            case TR_JNIVirtualTargetAddress:
+               {
+               uint8_t *startOfInstruction = self()->getBinaryEncoding();
+               uint8_t *startOfImmediate = cursor;
+               intptr_t diff = reinterpret_cast<intptr_t>(startOfImmediate) - reinterpret_cast<intptr_t>(startOfInstruction);
```

Comment:
did you want to add the < 256 assertion here as well as in OpenJ9 when the actual relocation record is created? I could understand if not (you *could* view as a requirement imposed by a downstream component) but given it's JNI and likely not to be reused by anything else, maybe catching a problem here is desirable? I'll leave it as optional, because I can see it both ways.

Reply:
I was thinking about it, but I think I mainly didn't add it here it so that there if, for whatever reason (perhaps on a different platform) the offset had to be increased to fit in a bigger type, then there would have to be an OMR PR just to update the assert. As such I figured it's probably best to just have the assert close to where the data in the relo header is actually stored.